src/google-classroom-extractor/edfi_google_classroom_extractor/mapping/assignment_submissions.py

```diff
@@ -63,6 +66,39 @@ def derive_state(submission_row: Series) -> str:
     return api_state
 
 
+def _get_submission_datetime(submission_row: Series) -> str:
+    if (
+        submission_row["state"] == TURNED_IN_STATE
+        or submission_row["state"] == RETURNED_STATE
```

Comment:
"Returned" means the teacher has graded the assignment. This isn't the submission date/time - should only be looking at `TURNED_IN_STATE`.

Reply:
What do you think about something like this?

If state == TURNED_IN { return updateTime }
if state == RETURNED { look in history and find the newest TURNED_IN entry }
else { it has not been turned in }

Reply:
✅

Reply:
I don't know if `If state == TURNED_IN { return updateTime }` is a good idea... What happens if a student turns in the submission and for some reason it gets updated after it without a change in the status? It is okay to only enter in the logic when the status is turned_in or returned, but we still may want to check the status history to have an accurate date